zcash_client_backend/src/welding_rig.rs

```diff
@@ -283,26 +265,38 @@ pub fn scan_block<P: consensus::Parameters, K: ScanningKey>(
                 })
                 .collect();
 
-            for (idx, c_out) in tx.outputs.into_iter().enumerate() {
+            for (index, output) in tx.outputs.into_iter().enumerate() {
                 // Grab mutable references to new witnesses from previous outputs
                 // in this transaction so that we can update them. Scoped so we
                 // don't hold mutable references to shielded_outputs for too long.
-                let mut new_witnesses: Vec<_> = shielded_outputs
+                let new_witnesses: Vec<_> = shielded_outputs
                     .iter_mut()
                     .map(|output| &mut output.witness)
                     .collect();
 
+                let output = CompactOutputDescription::try_from(output).ok().unwrap();
+
+                // Increment tree and witnesses
+                let node = Node::new(output.cmu.to_repr());
+                for witness in &mut *existing_witnesses {
```

Comment:
These two errors: ```error[E0382]: use of moved value: `existing_witnesses`
   --> zcash_client_backend/src/welding_rig.rs:243:32
    |
145 |     existing_witnesses: &mut [&mut IncrementalWitness<Node>],
    |     ------------------ move occurs because `existing_witnesses` has type `&mut [&mut IncrementalWitness<Node>]`, which does not implement the `Copy` trait
...
243 |                 for witness in existing_witnesses {
    |                                ^^^^^^^^^^^^^^^^^^ `existing_witnesses` moved due to this implicit call to `.into_iter()`, in previous iteration of loop
    |
note: this function takes ownership of the receiver `self`, which moves `existing_witnesses`
   --> /home/nuttycom/.rustup/toolchains/1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/collect.rs:234:18
    |
234 |     fn into_iter(self) -> Self::IntoIter;
    |                  ^^^^
help: consider creating a fresh reborrow of `existing_witnesses` here
    |
243 |                 for witness in &mut *existing_witnesses {
    |                                ++++++

For more information about this error, try `rustc --explain E0382`.
error: could not compile `zcash_client_backend` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
nuttycom@sword:~/work/librustzcash/canon (wallet_scan_batch_decrypt *)λ cargo check
   Compiling zcash_client_backend v0.5.0 (/home/nuttycom/work/librustzcash/canon/zcash_client_backend)
error[E0277]: `[&mut IncrementalWitness<Node>]` is not an iterator
   --> zcash_client_backend/src/welding_rig.rs:243:32
    |
243 |                 for witness in &mut existing_witnesses {
    |                                ^^^^^^^^^^^^^^^^^^^^^^^ `[&mut IncrementalWitness<Node>]` is not an iterator
    |
    = help: the trait `Iterator` is not implemented for `[&mut IncrementalWitness<Node>]`
    = note: required because of the requirements on the impl of `Iterator` for `&mut [&mut IncrementalWitness<Node>]`
    = note: 1 redundant requirements hidden
    = note: required because of the requirements on the impl of `Iterator` for `&mut &mut [&mut IncrementalWitness<Node>]`
    = note: required because of the requirements on the impl of `IntoIterator` for `&mut &mut [&mut IncrementalWitness<Node>]`
note: required by `into_iter`
   --> /home/nuttycom/.rustup/toolchains/1.56.1-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/iter/traits/collect.rs:234:5
    |
234 |     fn into_iter(self) -> Self::IntoIter;
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
I think that `existing_witnesses.iter_mut()` might have worked here instead, but it doesn't matter.